bindings/aws/s3/s3.go

```diff
@@ -92,13 +148,40 @@ func (s *AWSS3) parseMetadata(metadata bindings.Metadata) (*s3Metadata, error) {
 	return &m, nil
 }
 
-func (s *AWSS3) getClient(metadata *s3Metadata) (*s3manager.Uploader, error) {
+func (s *AWSS3) getReqParam(req *bindings.InvokeRequest, paramName string) (string, error) {
+	value := ""
+	err := (error)(nil)
+
+	if val, ok := req.Metadata[paramName]; ok && val != "" {
+		value = val
+	} else {
+		err = fmt.Errorf("param %s is required", paramName)
+	}
+
+	return value, err
+}
+
+func (s *AWSS3) getKey(req *bindings.InvokeRequest) (string) {
```

Comment:
only for CreateOperation should a new key be generated if not provided, for all other operations, it should through an error?

Reply:
+